src/components/tl/mlx5/tl_mlx5_team.c

```diff
@@ -136,33 +136,24 @@ ucc_status_t ucc_tl_mlx5_team_create_test(ucc_base_team_t *team)
         if (UCC_INPROGRESS == status) {
             return status;
         }
-        ucc_assert(status == UCC_OK);
         ucc_service_coll_finalize(tl_team->scoll_req);
-        if (tl_team->status[1] != UCC_OK) {
-            tl_debug(UCC_TL_TEAM_LIB(tl_team),
-                     "node leader failed during device memory init: %s",
-                     ucc_status_string(tl_team->status[1]));
-            ucc_tl_mlx5_team_destroy(team);
-            return tl_team->status[1];
-        }
         tl_team->state = TL_MLX5_TEAM_STATE_ALLTOALL_INIT;
     case TL_MLX5_TEAM_STATE_ALLTOALL_INIT:
-        status = ucc_tl_mlx5_team_alltoall_init_start(tl_team);
-        if (status != UCC_OK) {
-            tl_debug(UCC_TL_TEAM_LIB(tl_team), "failed to init a2a: %s",
-                     ucc_status_string(status));
-            return status;
-        }
+        tl_team->a2a_status = ucc_tl_mlx5_team_alltoall_init_start(tl_team);
         tl_team->state = TL_MLX5_TEAM_STATE_ALLTOALL_POSTED;
     case TL_MLX5_TEAM_STATE_ALLTOALL_POSTED:
-        status = ucc_tl_mlx5_team_alltoall_init_progress(tl_team);
-    }
-    if (status < 0) {
-        tl_debug(team->context->lib, "failed creating tl team: %p", tl_team);
-    } else if (status == UCC_OK) {
-        tl_debug(team->context->lib, "initialized tl team: %p", tl_team);
+        tl_team->a2a_status = ucc_tl_mlx5_team_alltoall_init_progress(tl_team);
+        if (tl_team->a2a_status == UCC_INPROGRESS) {
```

Comment:
what if the a2a_status is something other than UCC_OK and UCC_INPROGRESS? Can this situatino happen? also please do UCC_INPRGRESS == tl_team->a2a_status as mentioned before

Reply:
if the status is not `UCC_INPROGRESS`, it is simply stored in `team->a2a_status.local` at this stage. Then, at a2a's collective init, we will check this status and, in case of error, return "not supported" so that UCC can fallback to another TL.